test/PolyhedralGeometry/scalar_types.jl

```diff
@@ -15,11 +15,12 @@
   Qx, x = QQ[:x]
   K, r = number_field(x^3 - 3x^2 - 4x + 8, "r")
   Ky, y = K[:y]
-  L, = number_field(y^2 - (2 - r^2)//2, "q")
+  L, lq = number_field(y^2 - (2 - r^2)//2, "q")
   Lz, z = L[:z]
-  E, q = Hecke.embedded_field(L, real_embeddings(L)[2])
-  pq1 = E(roots(z^2 - (3 + 2r - r^2))[2])
-  pq2 = E(roots(z^2 - (3 - r^2))[1])
+  emb = only(filter(x -> is_positive(lq, x), real_embeddings(L)))
+  E, q = Hecke.embedded_field(L, emb)
+  pq1 = only(filter(x -> x > 0, E.(roots(z^2 - (3 + 2r - r^2)))))
+  pq2 = only(filter(x -> x > 0, E.(roots(z^2 - (3 - r^2)))))
```

Comment:
```suggestion
  pq1 = only(filter(is_positive, E.(roots(z^2 - (3 + 2r - r^2)))))
  pq2 = only(filter(is_positive, E.(roots(z^2 - (3 - r^2)))))
```
does this work as well?
If not, maybe replace by 
```suggestion
  pq1 = only(filter(>(0), E.(roots(z^2 - (3 + 2r - r^2)))))
  pq2 = only(filter(>(0), E.(roots(z^2 - (3 - r^2)))))
```
to eliminate some anonymous functions (with extra compilation needed)

Reply:
I only tried `ispositive` but `is_positive` does work. I also used it for the embeddings using `Base.Fix1` (which hopefully works on all relevant julia versions).